src/nft/hooks/usePurchaseAssets.ts

```diff
@@ -10,7 +10,8 @@ import { useTransactionResponse } from './useTransactionResponse'
 export function usePurchaseAssets(): (
   routingData: RouteResponse,
   assetsToBuy: UpdatedGenieAsset[],
-  purchasingWithErc20?: boolean
+  purchasingWithErc20?: boolean,
+  doNotInteractWithBag?: boolean
```

Comment:
nit: would name `doNotResetBag` to be more clear.

Also non-blocking but this is a good function to use [named parameters](https://betterprogramming.pub/named-parameters-in-typescript-e32c763d2b2e) so implementers can easily know the args they're passing. Will require a minor refactor.